test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('express')()
+const bodyParser = require('body-parser')
+const multer = require('multer') // v1.0.5
+const upload = multer() // for parsing multipart/form-data
+
+app.use(bodyParser.json()) // for parsing application/json
+app.use(bodyParser.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
+
+app.post('/profile', upload.array(), function (req, res, next) {
+  console.log(req.body)
+  res.json(req.body)
+})
+
+const port = 3001;
+
+app.get('/', (req, res) => res.send('Hello World!'));
+
+
+class Treenode {
```

Comment:
Something that would showcase code organization and class design is if you place this in its own module and import it where needed. This file while not that large could be more focused by clearing out the data modeling code and business logic to just focus on server + route initialization.